src/run/headers.ts

```diff
@@ -273,6 +274,7 @@ export const setCacheControlHeaders = (
     ['GET', 'HEAD'].includes(request.method) &&
     !headers.has('cdn-cache-control') &&
     !headers.has('netlify-cdn-cache-control') &&
+    !new URL(request.url).pathname.startsWith('/api/') &&
```

Comment:
we probably should not do this here and instead have something more proper for the API responses being potentially cached as it might impact not just pages router as this is trading one problem for another

It's fine to be there temporary while working on this PR, but should be removed before merging

Reply:
Removed in ca7ca4e0fc556b2bf5919a42e42abb4a5d16d104